### PR TITLE
Skip commit and push when no replacement happens in EventWatcher

### PR DESCRIPTION
**Pull request by t-kikuc**
**What this PR does**:

- Fix the following bug in EventWatcher by handling no-diff.

**Why we need it**:

- Without this PR, when no diff occurs by an event, the event will fail.

Before fix: `Failed to push changed files: err: exit status 128, out: fatal: invalid refspec ''`
image

After fix: `Nothing to commit`
image

- The cause is that when no diff occurs, `branchName` will be nil as below.

https://github.com/pipe-cd/pipecd/blob/d7f43699085c9da2ddcf913d8866d881cf8038c3/pkg/app/piped/eventwatcher/eventwatcher.go#L643-L645

**Which issue(s) this PR fixes**:

**Does this PR introduce a user-facing change?**:

- **How are users affected by this change**:
- **Is this breaking change**:
- **How to migrate (if breaking change)**:

### CODE REVIEW

**Review by khanhtc1202 — DISMISSED** [review]
Got your point, thanks 👍

File: pkg/app/piped/eventwatcher/eventwatcher.go [hunk 1]
```diff
@@ -388,7 +391,7 @@ func (w *watcher) execute(ctx context.Context, repo git.Repo, repoID string, eve
 			switch handler.Type {
 			case config.EventWatcherHandlerTypeGitUpdate:
 				branchName, err := w.commitFiles(ctx, latestEvent, matcher.Name, handler.Config.CommitMessage, e.GitPath, handler.Config.Replacements, tmpRepo, handler.Config.MakePullRequest)
-				if err != nil {
+				if err != nil && !errors.Is(err, errNoChanges) {
```
Comment (ffjlabo) [inline pkg/app/piped/eventwatcher/eventwatcher.go:394]: [IMO] How about using the bool variable like `change` to minimize the scope of the error variable and also for the readability?

```suggestion
				noChange := false
				if err != nil {
				    // 
				    if errors.Is(err, errNoChanges) {
				        noChange = true
				    } else {
				     ...   
				     continue
				    }
				}
				
				if noChange {
				        handledEvent.StatusDescription = "Nothing to commit"
					gitNoChangeEvents = append(gitNoChangeEvents, handledEvent)
				} else {
```

**Review by ffjlabo — COMMENTED** [review]
Thank you for the imporvement! I left an imo comment

File: pkg/app/piped/eventwatcher/eventwatcher.go (hunk 1 above)
Comment (t-kikuc) [reply pkg/app/piped/eventwatcher/eventwatcher.go]: thank you , got it, i fixed in [7b155e6](https://github.com/pipe-cd/pipecd/pull/5310/commits/7b155e62a7e7a994b97e6b4ac31a6d082cd415f5)

**Review by ffjlabo — APPROVED** [review]
LGTM

**Pull request merged**